src/models.py

```diff
@@ -25,6 +28,60 @@ def _verify_bbox_and_pages(self) -> None:
         if self.bounding_boxes is not None and self.pages is not None:
             assert len(self.bounding_boxes) == len(self.pages)
 
+    def merge(self, other: "Chunk", text_separator: str = " ") -> "Chunk":
```

Comment:
suggestion: could this be extended to merge >2 chunks? we do [a similar thing with spans in the KG](https://github.com/climatepolicyradar/knowledge-graph/blob/dc3c917cd1439d71718a1d02bf0892475843ef65/src/span.py#L125-L147) which started as single span, but the multi-span implementation has ended up being really useful!

Reply:
good shout! great to get some insight about what barriers this code could face a bit down the line :)